source/extensions/tracers/datadog/datadog_tracer_impl.cc

```diff
@@ -141,6 +146,22 @@ void TraceReporter::onSuccess(const Http::AsyncClient::Request& request,
   }
 }
 
+void TraceReporter::onClusterAddOrUpdate(Upstream::ThreadLocalCluster& cluster) {
```

Comment:
Can you move this into some kind of base class/mix-in shared by each tracer? I think it should be pretty easy to do.

Reply:
done.

I had to sacrifice `ENVOY_LOG(debug, ...)` statement, though.